### PR TITLE
Authentication Done

### PR DESCRIPTION
**Pull request by shawavisek35**
- Registration using firebase
- Login using firebase
- Minor bug fixes

### CODE REVIEW

**Review by supreethkurpad — APPROVED** [review]
Looks fine to me.

File: src/components/Header.tsx [hunk 1]
```diff
@@ -21,9 +35,16 @@ function Header () {
...
-            <li className="nav-item navs">
-              <a className="nav-link" href="/join">Register/Login</a>
-            </li>
+            {
+              fire.auth().currentUser != null
+                ? <li className="nav-item navs">
+                  <button onClick={() => logout()} className="nav-link logout">Logout</button>
+                </li>
+                : <li className="nav-item navs">
+                  <a className="nav-link" href="/join">Join</a>
```
Comment (bestinbthomas) [inline src/components/Header.tsx:44]:
```suggestion
                  <Link className="nav-link" to="/join">Join</a>
```
can you replace all `a` tags with `Link` tags

File: src/components/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          history.push('/')
+        })
+        .catch((err) => {
+          setUser({
+            ...user,
+            email: '',
+            password: ''
+          })
+          console.log(err)
+          toast.error('You are not registered. Try registering first')
```
Comment (bestinbthomas) [inline src/components/Login.tsx:40]: try getting the message from the response, there can be other reasons for failure like timeouts or any internal server errors

File: src/components/Register.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          })
+          .catch((err) => {
+            setUser({
+              ...user,
+              email: '',
+              password: '',
+              confirmPassword: ''
+            })
+            console.log(err)
+            toast.error('Server error occured')
```
Comment (bestinbthomas) [inline src/components/Register.tsx:46]: here also

**Review by bestinbthomas — CHANGES_REQUESTED** [review]
remove package-lock.json
use either npm or yarn not both

File: src/components/Header.tsx (hunk 1 above)
Comment (shawavisek35) [reply src/components/Header.tsx:44]: Done

File: src/components/Login.tsx (hunk 2 above)
Comment (shawavisek35) [reply src/components/Login.tsx:40]: Done

File: src/components/Register.tsx (hunk 3 above)
Comment (shawavisek35) [reply src/components/Register.tsx:46]: Done

**Comment by shawavisek35** [comment]
> remove package-lock.json
> use either npm or yarn not both

Hey I tried removing package-lock.json and entered it command yarn start to start the server but it showing error could not find package-lock.json. So am i missing anything?

**Comment by bestinbthomas** [comment]
> > remove package-lock.json
> > use either npm or yarn not both
> 
> Hey I tried removing package-lock.json and entered it command yarn start to start the server but it showing error could not find package-lock.json. So am i missing anything?

run
```
yarn
yarn start
```

**Review by bestinbthomas — APPROVED** [review]

**Pull request merged**